v1/src/main/java/com/google/cloud/teleport/templates/common/SpannerConverters.java

```diff
@@ -314,6 +357,42 @@ public void processElement(ProcessContext processContext) {
         processContext.output(read);
       }
 
+      /**
+       * Generate a column expression for use in a query based on user-supplied column and alias
+       * values.
+       *
+       * @param columnsToExport User-provided data in the format of "column1: alias1, column2,
+       *     column3: alias3".
+       * @return A prepared column expression suitable for use in a Data Query Language (DQL)
+       *     statement.
+       */
+      private String prepareColumnExpression(String columnsToExport) {
+        String[] columnsList = columnsToExport.split(",");
+        StringBuilder columnExpressionBuilder = new StringBuilder();
+
+        for (String columnWithAlias : columnsList) {
+          if (StringUtils.isBlank(columnWithAlias)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Please provide a valid comma-separated list of columns to be extracted. The received input is not valid: %s",
+                    columnsToExport));
+          }
+          // Split around ":" to check if alias is present
+          String[] columnMap = columnWithAlias.trim().split(":");
+          columnExpressionBuilder.append(columnMap[0].trim());
+          if (columnMap.length > 1) {
+            columnExpressionBuilder.append(" AS ");
+            columnExpressionBuilder.append(columnMap[1].trim());
+          }
+          columnExpressionBuilder.append(",");
+        }
+
+        // Removing the last character which is ','
+        columnExpressionBuilder.deleteCharAt(columnExpressionBuilder.length() - 1);
```

Comment:
Does this break if columnsToExport is empty? I understand we are checking that it is not empty before calling, But it is better to check if it is empty here. We might want to reuse this method later, better to make it handle edge cases.

Reply:
I think it's a valid point, I am making the method package-private from private so the check makes sense.